### PR TITLE
docker-sniffles2

### PR DESCRIPTION
**Pull request by 223050025**

### CODE REVIEW

File: .github/workflows/docker-publish-github.yml
Comment (Milokita) [inline .github/workflows/docker-publish-github.yml:1]: pls also add docker push to the dockerhub

File: entrypoint.sh
Comment (Milokita) [inline entrypoint.sh:1]: great

File: Dockerfile
Comment (Milokita) [inline Dockerfile:1]: nicely done

**Review by Milokita — CHANGES_REQUESTED** [review]

**Comment by 223050025** [comment]
1. Try to add Docker Hub login information and add Docker Hub push step.
2. Meet problem in log into docker hub: Error Username and password required.

@Milokita

File: .github/workflows/docker-publish-github.yml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+
+      - name: Build and push Docker image
+        id: build-and-push
+        uses: docker/build-push-action@0565240e2d4ab88bba5387d719585280857ece09
+        with:
+          context: .
+          push: ${{ github.event_name != 'pull_request' }}
+          tags: ${{ steps.meta.outputs.tags }}
```
Comment (Milokita) [inline .github/workflows/docker-publish-github.yml]: Tag : **latest** seems missing

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+# activate vir envir
+ENV PATH="/opt/venv/bin:$PATH"
+
+# Install Sniffles2 via pip
+RUN pip install sniffles
+
+# Set working directory
+WORKDIR /root
+
+# Add Bismark to PATH
```
Comment (Milokita) [inline Dockerfile]: Bismark ?

**Review by Milokita — CHANGES_REQUESTED** [review]

File: .github/workflows/docker-publish-github.yml (hunk 1 above)
Comment (223050025) [reply .github/workflows/docker-publish-github.yml]: 65 line: change to tags: cuhkhaosun/sniffles2:latest

Comment (223050025) [reply .github/workflows/docker-publish-github.yml]: Merge Docker build steps and push both to GitHub Container Registry (ghcr.io) and to Docker Hub

**Comment by 223050025** [comment]
Still problem: log into docker hub: Error Username and password required.
@Milokita

**Review by Milokita — APPROVED** [review]

**Pull request merged**